modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase

```diff
@@ -749,6 +749,126 @@ definition {
+					fieldLabelName = "Field Text ${letters}",
+					fieldName = "customField",
+					fieldType = "String",
+					isRequired = "false",
+					objectName = "CustomObject${letters}182607");
+
+				ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject${letters}182607");
+			}
+		}
+
+		task ("And given a 1:M relationship between the Object A and Object B") {
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "CustomObjectA182607",
+				objectName_2 = "CustomObjectB182607",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given a User Custom with admin roles is added") {
+			JSONUser.addUser(
+				userEmailAddress = "userea@liferay.com",
+				userFirstName = "userfn",
+				userLastName = "userln",
+				userScreenName = "usersn");
+
+			JSONRole.assignRoleToUser(
+				roleTitle = "Administrator",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("And given a user notification template is created using terms from both object definitions") {
+			UserNotificationAPI.postNotificationTemplate(
+				en_US = "[%RELATIONSHIP_CUSTOMOBJECTA182607_CUSTOMFIELD%] | [%CUSTOMOBJECTB182607_CUSTOMFIELD%]",
+				externalReferenceCode = "ercTest",
+				name = "User Notification Template",
+				recipientType = "user",
+				type = "userNotification",
+				userScreenName = "usersn");
+		}
+
+		task ("And given the title field of Object A is changed") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectSystemObject(label = "Custom Object A 182607");
+
+			CreateObject.selectTitleField(fieldLabel = "Field Text A");
+
+			CreateObject.saveObject();
+		}
+
+		task ("And given a notification action is created on the Object 2 using the on after add trigger") {
+			Navigator.gotoBack();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object B 182607");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.addObjectActionViaUI(
+				actionLabel = "Notification action when entry is added",
+				active = "true",
+				notificationName = "User Notification Template",
+				objectLabel = "Custom Object B 182607",
+				thenAction = "Notification",
+				whenAction = "On After Add");
+		}
+
+		task ("And given the User is changed for the new User Custom") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
+		}
+
+		task ("When an object entry for both Objects") {
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "customField",
+				objectName = "CustomObjectA182607",
+				value = "Entry Test A");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObjectB182607");
+
+			LexiconEntry.gotoAdd();
+
+			Type(
+				key_fieldLabel = "Field Text B",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "Entry Test B");
+
+			ObjectPortlet.searchEntryRelationshipField(entryValue = "Entry Test A");
+
+			PortletEntry.save();
+		}
```

Comment:
I added the second entry via UI because the value and relationship needed to be added at the same time to replace the entry term used in the notification, and that isn't possible via the API.